node/subsystem-util/src/determine_new_blocks.rs

```diff
@@ -0,0 +1,517 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A utility for fetching all unknown blocks based on a new chain-head hash.
+
+use polkadot_node_subsystem::{
+	messages::ChainApiMessage,
+	SubsystemSender, SubsystemError, SubsystemResult,
+};
+use polkadot_primitives::v1::{Hash, Header, BlockNumber};
+use futures::prelude::*;
+use futures::channel::oneshot;
+
+/// Given a new chain-head hash, this determines the hashes of all new blocks we should track
+/// metadata for, given this head. The list will typically include the `head` hash provided unless
+/// that block is already known, in which case the list should be empty. This is guaranteed to be
+/// a subset of the ancestry of `head`, as well as `head`, starting from `head` and moving
+/// backwards.
+///
+/// This returns the entire ancestry up to the provided lower bound height or the last item we
+/// have in the DB. This may be somewhat expensive when first recovering from major sync.
```

Comment:
nits:
1. up to (not including)
2. we have in the DB -> that is known? since we removed the db argument